contracts/src/components/attack.cairo

```diff
@@ -0,0 +1,63 @@
+use eternum::alias::ID;
+use eternum::utils::coin_toss;
+
+#[derive(Component, Copy, Drop, Serde, SerdeLen)]
+struct Attack {
+    #[key]
+    entity_id: ID,
+    chance: u8,
```

Comment:
don't think we should call this chance, just value or something to make it generic even though we currently we do use it as proability in the attack system